analyzers/src/SonarAnalyzer.CSharp/Rules/AssertionArgsShouldBePassedInCorrectOrder.cs

```diff
@@ -18,63 +18,50 @@
+    {
+        ["AreEqual"]    = ImmutableArray.Create(KnownType.Microsoft_VisualStudio_TestTools_UnitTesting_Assert, KnownType.NUnit_Framework_Assert),
+        ["AreNotEqual"] = ImmutableArray.Create(KnownType.Microsoft_VisualStudio_TestTools_UnitTesting_Assert, KnownType.NUnit_Framework_Assert),
+        ["AreSame"]     = ImmutableArray.Create(KnownType.Microsoft_VisualStudio_TestTools_UnitTesting_Assert, KnownType.NUnit_Framework_Assert),
+        ["AreNotSame"]  = ImmutableArray.Create(KnownType.Microsoft_VisualStudio_TestTools_UnitTesting_Assert, KnownType.NUnit_Framework_Assert),
+        ["Equal"]       = ImmutableArray.Create(KnownType.Xunit_Assert),
+        ["Same"]        = ImmutableArray.Create(KnownType.Xunit_Assert),
+        ["NotSame"]     = ImmutableArray.Create(KnownType.Xunit_Assert)
+    };
 
-                var symbolInfo = c.SemanticModel.GetSymbolInfo(methodCallExpression.Expression).Symbol;
-                var isAnyTrackedAssertType = (symbolInfo as INamedTypeSymbol).IsAny(methodKnownTypes);
-                if (!isAnyTrackedAssertType)
-                {
-                    return;
-                }
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterNodeAction(c =>
+        {
+            if (c.Node is InvocationExpressionSyntax { ArgumentList: { Arguments: { Count: >= 2 } arguments } argumentList, Expression: MemberAccessExpressionSyntax methodCallExpression }
+                && methodCallExpression.IsKind(SyntaxKind.SimpleMemberAccessExpression)
+                && CSharpFacade.Instance.MethodParameterLookup(argumentList, c.SemanticModel) is var parameterLookup
+                && CheckArguments(parameterLookup)
+                && MethodsWithType.GetValueOrDefault(methodCallExpression.Name.Identifier.ValueText) is { } methodKnownTypes
+                && methodKnownTypes.IsDefault is false
+                && (c.SemanticModel.GetSymbolInfo(methodCallExpression.Expression).Symbol as INamedTypeSymbol).IsAny(methodKnownTypes))
+            {
+                c.ReportIssue(Diagnostic.Create(Rule, arguments[0].CreateLocation(arguments[1])));
+            }
+        },
+        SyntaxKind.InvocationExpression);
 
-                c.ReportIssue(Diagnostic.Create(Rule, firstArgument.CreateLocation(secondArgument)));
-            },
-            SyntaxKind.InvocationExpression);
-    }
+    private bool CheckArguments(IMethodParameterLookup parameterLookup) =>
+        // "notExpected" is used in MSTest's AreNotEqual and AreNotSame
+        (parameterLookup.TryGetSyntax("expected", out var expected) || parameterLookup.TryGetSyntax("notExpected", out expected))
+        && expected.FirstOrDefault() is not LiteralExpressionSyntax
+        && parameterLookup.TryGetSyntax("actual", out var actual)
+        && actual.FirstOrDefault() is LiteralExpressionSyntax;
```

Comment:
We could also consider using [`semanticModel.GetConstantValue`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.semanticmodel.getconstantvalue?view=roslyn-dotnet-4.3.0) instead. This would also detect
```cs
const int expected = 1;
Assert.Equal(someInt, expected);
```